numba/core/callconv.py

```diff
@@ -580,10 +581,10 @@ def emit_unwrap_dynamic_exception_fn(self, module, st_type, nb_types):
             val = builder.extract_value(builder.load(st_ptr), i)
             obj = pyapi.from_native_value(typ, val)
 
-            # not all objects are supported
+            # If object cannot be boxed, raise an exception
             if obj == cgutils.get_null_value(obj.type):
                 # When not supported, abort compilation
-                msg = f'cannot convert native {typ} to python object'
+                msg = f'Cannot convert native {typ} to a Python object.'
```

Comment:
There is no unit test for this condition?

Reply:
Yes, there is. See `test_exceptions::TestRaising::test_dynamic_raise_bad_args`

Reply:
Ah, thanks, apologies, completely missed that (I skimmed the diff of the last two commits for two updates and missed the second in the test suite).